recbole/model/general_recommender/ract.py

```diff
@@ -0,0 +1,251 @@
+        return rating_matrix
+
+    def mlp_layers(self, layer_dims):
+        mlp_modules = []
+        for i, (d_in, d_out) in enumerate(zip(layer_dims[:-1], layer_dims[1:])):
+            mlp_modules.append(nn.Linear(d_in, d_out))
+            if i != len(layer_dims[:-1]) - 1:
+                mlp_modules.append(nn.Tanh())
+        return nn.Sequential(*mlp_modules)
+
+    def reparameterize(self, mu, logvar):
+        if self.training:
+            std = torch.exp(0.5 * logvar)
+            epsilon = torch.zeros_like(std).normal_(mean=0, std=0.01)
+            return mu + epsilon * std
+        else:
+            return mu
+
+    def forward(self, rating_matrix):
+
+        t = F.normalize(rating_matrix)
+
+        h = F.dropout(t, self.drop_out, training=self.training) * (1 - self.drop_out)
+        self.input_matrix = h
+        self.number_of_seen_items = (h != 0).sum(dim=1)             # network input
+
+        mask = (h > 0) * (t > 0)
+        self.true_matrix = t * ~mask
+        self.number_of_unseen_items = (self.true_matrix != 0).sum(dim=1) # remaining input
+
+        h = self.encoder(h)
+
+        mu = h[:, :int(self.lat_dim / 2)]
+        logvar = h[:, int(self.lat_dim / 2):]
+
+        z = self.reparameterize(mu, logvar)
+        z = self.decoder(z)
+        self.predict_matrix = z
+        return z, mu, logvar
+
+    def calculate_actor_loss(self, interaction):
+
+        user = interaction[self.USER_ID]
+        rating_matrix = self.get_rating_matrix(user)
+
+        self.update += 1
+        if self.total_anneal_steps > 0:
+            anneal = min(self.anneal_cap, 1. * self.update / self.total_anneal_steps)
+        else:
+            anneal = self.anneal_cap
+
+        z, mu, logvar = self.forward(rating_matrix)
+
+        # KL loss
+        kl_loss = -0.5 * (torch.sum(1 + logvar - mu.pow(2) - logvar.exp(), dim=1)) * anneal
+
+        # CE loss
+        ce_loss = -(F.log_softmax(z, 1) * rating_matrix).sum(1)
+
+        return ce_loss + kl_loss
+
+    def construct_critic_input(self, actor_loss):
+        critic_inputs = []
+        critic_inputs.append(self.number_of_seen_items)
+        critic_inputs.append(self.number_of_unseen_items)
+        critic_inputs.append(actor_loss)
+        return torch.stack(critic_inputs, dim=1)
+
+    def construct_critic_layers(self):
+        mlp_modules = []
+        mlp_modules.append(nn.BatchNorm1d(3))
+        mlp_modules.append(nn.Linear(3, self.critic_layers_1))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_1, self.critic_layers_2))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_2, self.critic_layers_3))
+        mlp_modules.append(nn.ReLU())
+        mlp_modules.append(nn.Linear(self.critic_layers_3, 1))
+        mlp_modules.append(nn.Sigmoid())
+        return nn.Sequential(*mlp_modules)
```

Comment:
Maybe `critic_layers_1/2/3` can be summarized into one single arg (e.g. [100, 100, 10]).

Then we can use a loop to construct critic layers.